lib/ansible/galaxy/token.py

```diff
@@ -35,6 +48,93 @@
 
 display = Display()
 
+AH_TOKEN_EXPIRES_N_MIN_EARLY = 1
+AH_AUTH_URL = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+AH_JWKS_URI = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/certs"
+
+# https://www.rfc-editor.org/rfc/rfc7518#section-3.1
+REQUIRED_JWT_HEADER_KEYS = set(['alg', 'typ', 'kid'])
+REQUIRED_JWT_HEADER_VALUES = {
+    'alg': ['RS256'],
+    'typ': ['JWT'],
+}
+
+PublicNumbers = namedtuple("PublicNumbers", ["e", "n"])
+
+
+def validate_jwt_format(data):
+    if len(data.split('.')) != 3:
```

Comment:
Creating an in-memory list only to count its elements is probably more expensive than this:
```suggestion
    if data.count('.') != 3:
```